File: chain-of-responsibility/typescript/image.ts
// Uygulama süresince resimlerimizi temsil edecek sınıf.
export class Image {
  public name: string;
  public extension: string;

  constructor(name: string, extension: string) {
    this.name = name;
    this.extension = extension;
  }
}

File: chain-of-responsibility/go/image.go
package main

// Image uygulama süresince resimlerimizi temsil edecek sınıf.
type Image struct {
	Name      string
	Extension string
}

func NewImage(name string, extension string) *Image {
	return &Image{
		Name:      name,
		Extension: extension,
	}
}
